src/index.js

```diff
@@ -88,4 +103,18 @@ export default class Zeroconf extends EventEmitter {
     RNZeroconf.stop()
   }
 
+  /**
+   * Publish a service
+   */
+  publishService(domain = 'local.', type, name, port) {
```

Comment:
Advertising should take a string for "type" and a string for protocol, then prefix them with underscores and join them together, just so it's consistent with the scan.

Whilst I'd love to do this, I'm not an iOS or Android developer, and all the changes needed to achieve this are in the bits for each platform (if we're to be consistent).